[arnav-goel10] iP

arnav-goel10 opened the pull request:
# ~~Deadpool~~ Darkpool

> “Let’s make this PR so good, even your code will be impressed with itself.” – Deadpool

Welcome to Darkpool, the task management tool that’s so sleek, it might just have its own fan club. Here’s why you’ll love it:

- 😇 **Ridiculously user-friendly**
- ⚡ **Faster than a speeding bullet**

Here’s how to get rolling:

1. **Clone** the [repo](https://github.com/arnav-goel10/ip) like a pro.
2. **Build** it with Maven or Gradle—your choice!
3. **Run** it and bask in the glory of perfect task management.
4. **Revel** in your newfound productivity!

### Features:

- [x] Task management that doesn’t suck
- [ ] Deadline tracking (coming soon, like a sequel)
- [ ] Custom reminders (also coming soon, like a sequel to a sequel)

For those who like a bit of coding with their caffeine, here’s the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(DarkpoolApp.class, args);
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

arnav-goel10 committed:
Changed Duke to Darkpool

arnav-goel10 committed:
Change text colour

arnav-goel10 committed:
Add echo

arnav-goel10 committed:
Save user data

arnav-goel10 committed:
Add task class

arnav-goel10 committed:
Add isDone to task

arnav-goel10 committed:
Fix Task toString

arnav-goel10 committed:
Implement Task in Darkpool

arnav-goel10 committed:
Update Readme

arnav-goel10 committed:
Update Readme again

arnav-goel10 committed:
Update Readme again

arnav-goel10 committed:
Add Todo class

arnav-goel10 committed:
Add Deadline class

arnav-goel10 committed:
Add event class

arnav-goel10 committed:
Made Task class abstract

arnav-goel10 committed:
Implement todo, task and deadline in Darkpool

arnav-goel10 committed:
Error with expected.txt

arnav-goel10 committed:
Test

arnav-goel10 committed:
Handle Errors

arnav-goel10 committed:
Error handle

arnav-goel10 committed:
Add space

arnav-goel10 committed:
A-TextUiTesting

arnav-goel10 committed:
Add save

arnav-goel10 committed:
Merge branch-level-7

arnav-goel10 committed:
Implement Date in Event and Deadline

arnav-goel10 committed:
Merge with branch-level-8

arnav-goel10 committed:
OOP

arnav-goel10 committed:
Implement OOP

arnav-goel10 committed:
Made Packages

arnav-goel10 committed:
Merge Gradle

arnav-goel10 committed:
Add J-Unit

arnav-goel10 committed:
Level-6

arnav-goel10 committed:
Merge gradle branch

arnav-goel10 committed:
Fix ClassName

arnav-goel10 committed:
Add javadocs

arnav-goel10 committed:
Add javadocs

arnav-goel10 committed:
Add javadocs

arnav-goel10 committed:
Fix coding standard

arnav-goel10 committed:
Merge master with coding standard

Meowloid commented on src/main/java/darkpool/util/Storage.java:
```diff
@@ -0,0 +1,127 @@
+     * @return The file object representing the data file.
+     * @throws DarkpoolException If an I/O error occurs.
+     */
+    private File fileAndDirCheck() throws DarkpoolException {
```
I think the name of this method can be changed to fit what W4.6b recommends.
![image](https://github.com/user-attachments/assets/fb508669-48c7-431e-89d1-aabbc79a8caf)

no hard feelings, just tryna get those participation marks yk

arnav-goel10 replied:
Thanks for pointing the mistake out! I have fixed it.

arnav-goel10 replied:
I have updated it to validateData. Thanks for pointing it out.

Meowloid commented on src/main/java/darkpool/util/Ui.java:
```diff
@@ -0,0 +1,120 @@
+     * Displays the greeting message.
+     */
+    public void greeting() {
+        upperLine();
```
I like the use of methods and predefined Strings to help print Ui elements.

arnav-goel10 replied:
thanks!

Meowloid commented on src/test/java/darkpool/task/DeadlineTest.java:
```diff
@@ -0,0 +1,64 @@
+    private String invalidTime;
+    private String description;
+
+    @BeforeEach
```
I like the concise testing.

Meowloid reviewed (commented):
LGTM. Well done 👍 Love the consistent snarky character throughout.

arnav-goel10 committed:
Change method name

arnav-goel10 committed:
Implement find command

arnav-goel10 committed:
Update method name

arnav-goel10 committed:
Fix checkstyle errors

arnav-goel10 committed:
Fix

bbryant824 commented on README.md:
````diff
@@ -7,18 +7,54 @@ This is a project template for a greenfield Java project. It's named after the J
-   |____/ \__,_|_|\_\___|
-   ```
+4. After that, locate the `src/main/java/Duke.java` file, right-click it, and choose `Run Duke.main()` (if the code editor is showing compile errors, try restarting the IDE). If the setup is correct, you should see something like the below as the output:
+
````
very cool, very awesome way to start the chatbot!!!

bbryant824 commented on src/main/java/darkpool/Darkpool.java:
```diff
@@ -0,0 +1,70 @@
+ * The main class for the Darkpool application.
+ * It initializes the necessary components and handles the main program loop.
+ */
+public class Darkpool {
```
The class name Darkpool follows the naming convention for classes (PascalCase) and is appropriately descriptive. However, you could consider adding a more descriptive name like DarkpoolApplication if Darkpool doesn't sufficiently reflect the purpose of the class in a larger context.

bbryant824 commented on src/main/java/darkpool/util/Parser.java:
```diff
@@ -0,0 +1,201 @@
+        if (fromParts.length > 2) {
+            throw new DarkpoolException("only one from pls");
+        } else if (fromParts.length < 2 || fromParts[1].trim().isEmpty()) {
+            throw new DarkpoolException("where is from bruh");
```
Consider aligning the error messages to be more professional, unless the informal tone is a deliberate stylistic choice for the application. And in this case, to me, I think it is quite cool to sound this way.

bbryant824 commented on src/main/java/darkpool/util/Parser.java:
```diff
@@ -0,0 +1,201 @@
+        try {
+            num = Integer.parseInt(userInput[1]) - 1;
+        } catch (NumberFormatException e) {
+            throw new DarkpoolException("do you know what a number is?");
```
This is quite funny, I like it although it's a bit not user-friendly.

bbryant824 commented on src/test/java/darkpool/util/TaskListTest.java:
```diff
@@ -0,0 +1,119 @@
+package darkpool.util;
+
```
The indentation is consistent throughout the file, with proper 4-space indentation. This improves readability and follows standard Java formatting practices.  very good!

bbryant824 reviewed (commented):
You generally follow good naming conventions with meaningful, descriptive names for variables, methods, and classes. Your use of camelCase for methods and variables, and PascalCase for classes is consistent and aligns with Java’s coding standards. Very creative also, adding the fun elements into the chatbot, very impressive.

arnav-goel10 committed:
Add Gui

arnav-goel10 committed:
Add gui

arnav-goel10 committed:
Fix build.gradle

arnav-goel10 committed:
Update Javadocs for Darkpool, Launcher, Gui, Parser, Storage

Enhanced Javadocs for the following classes:
Darkpool: Improved descriptions for methods and added details on
functionality and parameters.
Launcher: Updated documentation to clarify startup procedures and
configuration options.
Gui: Revised Javadocs to better explain user interface components and
interactions.
Parser: Added explanations for parsing logic and error handling
procedures.
Storage: Corrected and expanded documentation on data storage
mechanisms and usage.

These updates ensure that the Javadocs provide clear and accurate
information, facilitating better understanding and maintenance of the
codebase.

arnav-goel10 committed:
Add assert statements to Darkpool and Parser

Introduced multiple assert statements to enhance runtime validation:
- Added assertions to ensure non-null responses after command execution
  and processing.
- Included checks for expected conditions and data integrity.

These assertions improve code reliability by validating critical conditions
and aiding in debugging and error detection.

arnav-goel10 committed:
Add assert statement to darkpool

arnav-goel10 committed:
Split parser class into smaller classes

arnav-goel10 committed:
Divide tasklist class into smaller classes

arnav-goel10 committed:
Split storage class to smaller classes

arnav-goel10 committed:
Split into different packages

arnav-goel10 committed:
Merge master with A-Assertions

arnav-goel10 committed:
Remove previous parser

arnav-goel10 committed:
Merge master with a-codequality

arnav-goel10 committed:
Implement B-DoAfterTasks

arnav-goel10 committed:
Fix test cases for new code

arnav-goel10 committed:
Update readme file

arnav-goel10 committed:
Add javadocs to storage, tasklist and parser

arnav-goel10 committed:
Merge pull request #2 from arnav-goel10/UserGuide

Add javadocs to storage, tasklist and parser

arnav-goel10 committed:
Fix checkstyle errors

arnav-goel10 committed:
Merge with userguide branch

arnav-goel10 committed:
Add workflow to project

arnav-goel10 committed:
Add title to main window

arnav-goel10 committed:
Add Ui.png to docs folder

arnav-goel10 committed:
Update readme file

arnav-goel10 committed:
Update Ui.png image

arnav-goel10 committed:
Change readme file again

arnav-goel10 committed:
Add config file

arnav-goel10 committed:
Fix config file

arnav-goel10 committed:
Fix link in readme

arnav-goel10 committed:
Fix for github pages

arnav-goel10 committed:
Remove redundant code

arnav-goel10 committed:
Update build.gradle file

arnav-goel10 committed:
Update build.gradle

arnav-goel10 committed:
Update build gradle file

arnav-goel10 committed:
Fix header comments
